src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs

```diff
@@ -98,14 +127,14 @@ public virtual IApplication Start()
             _applicationLifetime.NotifyStarted();
             logger.Started();
 
-            return new Application(ApplicationServices, Server.Features, new Disposable(() =>
-            {
-                logger.Shutdown();
-                _applicationLifetime.StopApplication();
-                Server.Dispose();
-                _applicationLifetime.NotifyStopped();
-                (_applicationServices as IDisposable)?.Dispose();
-            }));
+            return new Disposable(() =>
```

Comment:
```
        public static void Main(string[] args)
        {
            var config = WebApplicationConfiguration.GetDefault(args);

            var builder = new WebApplicationBuilder()
                .UseConfiguration(config)
                .UseStartup<StartupBlockingOnStart>();

            using (var application = builder.Build())
            {
                application.Start();
                Console.ReadLine();
            }
        }
```

This is what usage looks like for the alternative API where we `Start()` returns `void`. Does this look better than the current usage?